otelcol/configprovider.go

```diff
@@ -142,12 +144,3 @@ func (cm *configProvider) Watch() <-chan error {
 func (cm *configProvider) Shutdown(ctx context.Context) error {
 	return cm.mapResolver.Shutdown(ctx)
 }
-
-func (cm *configProvider) GetConfmap(ctx context.Context) (*confmap.Conf, error) {
```

Comment:
Should we keep a dummy implementation of this method for one or more releases that returns an error saying that this functionality is deprecated?

Reply:
I removed this because I don't expect anyone else is using it, but thinking on it, this should go through the normal deprecation process. I've added a deprecation notice to this method and we can just remove the implementation when we remove the interface. I would feel comfortable removing both one release after deprecating them.